testing/trino-testing/src/main/java/io/trino/testing/QueryAssertions.java

```diff
@@ -94,6 +100,51 @@ else if (count.isPresent()) {
         }
     }
 
+    private static void assertDistributedUpdate(DistributedQueryRunner distributedQueryRunner, Session session, @Language("SQL") String sql, OptionalLong count, Optional<Consumer<Plan>> planAssertion)
```

Comment:
I don't see a good way to get a `QueryId` out of `LocalQueryRunner` that would allow these code paths to be shared unfortunately, but if you see a way to do that let me know and I'll make that change. Agreed on the duplication here being less ideal.